src/Core/src/FileStream.cpp

```diff
@@ -0,0 +1,129 @@
+#include "FileStream.h"
+#include <algorithm>
+#include <stdexcept>
+
+namespace OpenLoco
+{
+    static size_t getFileLength(std::fstream& fs)
+    {
+        auto backup = fs.tellg();
+        fs.seekg(0, std::ios_base::end);
+        auto len = fs.tellg();
+        fs.seekg(backup);
+        return len;
+    }
+
+    FileStream::FileStream(const std::filesystem::path& path, StreamMode mode)
+    {
+        open(path, mode);
+    }
+
+    bool FileStream::open(const std::filesystem::path& path, StreamMode mode)
+    {
+        if (mode == StreamMode::none)
+        {
+            throw std::invalid_argument("Invalid mode argument");
+        }
+
+        close();
+
+        if (mode == StreamMode::write)
+        {
+            _fstream.open(path, std::ios::out | std::ios::binary);
+        }
+        if (mode == StreamMode::read)
+        {
+            _fstream.open(path, std::ios::in | std::ios::binary);
+        }
+
+        if (!_fstream.is_open())
+        {
+            return false;
```

Comment:
I think we might need to throw still here until all consumers check the result.

Reply:
Ah yeah, damn. I think I can only have that in the constructor as open didn't exist before that.